### DISCUSSION TITLE
Improved tooltip display for TypedDicts

### DISCUSSION THREAD

**Discussion by stereobutter** [microsoft/pylance-release#2674] [Enhancement]
Since `TypedDict` typecheck with dict literals (and they are not classes at runtime) it would be helpful if the tooltip for subclasses of `TypedDict` could be improved to reflect this. Take for example

<img width="429" alt="example of TypedDict Tooltip" src="https://user-images.githubusercontent.com/35063695/149987370-2fd5c22a-063c-4dcc-b663-83faa6dc673a.png">

where  `class Example` is somewhat misleading and a more helpful tooltip would be something like `TypedDict Example of {'a': int, 'b': str}`

**Comment by erictraut** [comment]
There are many concepts in the Python type system that act like classes but are not classes at runtime. Examples include symbols defined via `NewType`, type aliases that reference classes, etc. All of these are assignable to `type` or `Type[T]`. As such, I don't know that it makes sense to single out `TypedDict` as its own category of symbols when it comes to hover text.

**Comment by stereobutter** [comment]
What imho. makes `TypedDict` special (compared to say `NewType`) is that good support for it brings quality of life improvements for non typing users as well (similar arguments have been made on https://github.com/microsoft/pylance-release/issues/654). Especially when `TypedDict` is used with e.g. `pydantic` for configuration autocomplete + type checking + intuitive tooltips would make this a very nice user experience.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
